src/cephadm/cephadm.py

```diff
@@ -483,14 +483,51 @@ def daemon_name_or_type(daemon: Dict[str, str]) -> str:
     if by_name and '.' not in daemon_filter:
         logger.warning(f'Trying to get container info using invalid daemon name {daemon_filter}')
         return None
-    daemons = list_daemons(ctx, detail=False)
-    matching_daemons = [d for d in daemons if daemon_name_or_type(d) == daemon_filter and d['fsid'] == ctx.fsid]
+    if by_name:
+        matching_daemons = list_daemons(ctx, daemon_name=daemon_filter)
+        if len(matching_daemons) > 1:
+            logger.error(f'Found multiple daemons sharing same name: {daemon_filter}')
+            # Take the first daemon we find that is actually running, or just the
+            # first in the list if none are running
+            matched_daemon = None
+            for d in matching_daemons:
+                if 'state' in d and d['state'] == 'running':
+                    matched_daemon = d
+                    break
+            if not matched_daemon:
+                matched_daemon = matching_daemons[0]
+            matching_daemons = [matched_daemon]
+    else:
+        daemons = list_daemons(ctx, detail=False)
```

Comment:
In the other branch detail is not set to anything. Are they intended to be different?

Reply:
yes, `detail=False` excludes the container image field from the result. in the case where the container is running, that is okay, but when it isn't we need that field.

Reply:
OK, can you make that comment in the code, please?